docs/gfx-user-guide/01-getting-started.md

```diff
@@ -0,0 +1,342 @@
+
+### Install Header Files
+Once you have built or obtained a Slang release, make the following header files from the release package accessible to your application:
+- `slang-gfx.h`
+- `slang.h`
+- `slang-com-ptr.h`
+- `slang-com-helper.h`
+
+### Linking the Library
+On Windows (with `msvc`), make sure that `gfx.lib` is provided as linker input via the `Linker->Input->Additional Dependencies` project configuration. On Unix systems, make sure to pass `-lgfx` when compiling your application.
+
+Creating a GPU Device
+---------------------------
+
+To start using the graphics layer, create an `IDevice` object by calling `gfxCreateDevice`. The `IDevice` interface is the main entry-point to interact with the graphics layer. It represent GPU device context where all interactions with the GPU take place.
+
+```C++
+#include "slang-gfx.h"
+
+using namespace gfx;
+
+IDevice* gDevice = nullptr;
+
+void initGfx()
+{
+    IDevice::Desc deviceDesc = {};
+    gfxCreateDevice(deviceDesc, &gDevice);
+}
+```
+
+The `IDevice::Desc` struct passed to `gfxCreateDevice` defines many configurations on how a device shall be created. Most notably, the `deviceType` field specifies what underlying graphics API to use. By default, `gfxCreateDevice` will attempt to use the best API available on current platform. On Windows, the layer will prefer to use `D3D12` but will also try to use `Vulkan`, `D3D11`, `OpenGL` in order, in case the former API isn't available. On Unix systems, it will always default to `Vulkan` since this is the only API that supports full Graphics capabilities. A user can always specify the `deviceType` field to force the layer to use a specific API. If the device creation succeeds, `gfxCreateDevice` will return `SLANG_OK(0)`.
+
+Similar to the Slang API, objects created by the graphics layer also conforms to the COM standard. The user to responsible for calling `release` method on every object returned to the user by the layer to prevent memory leaks.
+
+Enabling the Debug Layer
+--------------------------
+
+The Slang Graphics Layer provides a debug layer that can be enabled to perform additional validations to ensure correctness. To enable the debug layer, simply call `gfxEnableDebugLayer` before calling `gfxCreateDevice`.
+
+To receive diagnostic messages, you need to create a class that implements the `IDebugCallback` interface, and call `gfxSetDebugCallback` to provide the callback instance to the graphics layer. For example:
+
+```C++
+struct MyDebugCallback : public IDebugCallback
+{
+    virtual SLANG_NO_THROW void SLANG_MCALL handleMessage(
+        DebugMessageType type,
+        DebugMessageSource source,
+        const char* message) override
+    {
+        printf("%s\n", message);
+    }
+};
+
+MyDebugCallback gCallback;
+void initGfx()
+{
+    gfxEnableDebugLayer();
+    gfxSetDebugCallback(&gCallback);
+
+    IDevice::Desc deviceDesc = {};
+    gfxCreateDevice(&deviceDesc, &gDevice);
+}
+```
+
+Creating a Pipeline State
```

Comment:
We probably need/want to iterate on the ordering a bit here. This flow jumps right into creating a pipeline state before setting up queues or any of that stuff. I would have expected the flow to be more organized around the order in which an application needs to initialize/create the objects at runtime (you probably create your queue(s) shortly after creating your device, etc.).

The code blocks here also seem to go a bit long, so we will eventually want to edit things to have more prose in between to explain what is going on.

In the specific case of the pipeline state creation here, I might argue that we want to leave out the issue of how the user creates a `slang::IComponentType` to represent their program, at least at first. If we assume that the user already has the component type, then it becomes much easier to show how to create a pipeline state, since it is almost trivial for the compute case.